src/AddressBook.java

```diff
@@ -41,13 +41,17 @@ public void remove(int index) {
 
     /**
      * Sets the person at the given index to the Person specified.
+     * Code changed to add person !=null to fix bug
      *
      * @param index  The table index of the person to update.
      * @param person Person to replace index location with.
      */
     public void set(int index, Person person) {
-        persons.set(index, person);
-        fireTableRowsUpdated(index, index);
+        if (person!=null){
+            persons.set(index, person);
+            fireTableRowsUpdated(index, index);
+        }
+
```

Comment:
Added this statement, because a bug occurs:
 if you add a valid person, and set with invalid input.
It causes a null person to be added to address book and the whole program begins to lag and become unfunctional

Reply:
Good spot.